### PR TITLE
change to utest runner

### PR DESCRIPTION
**Pull request by lrq619**
## Summary

PMU tools can not work normally on github hosted runners
Add a self hosted runner and assign profile unit test running on that runner

## Implementation Notes :hammer_and_pick:

* Change the implementation of detecting pmu tools and perf: stdout cannot be captured in self-hosted runners

## External Dependencies :four_leaf_clover:

* N/A

## Breaking API Changes :warning:

* N/A

*Simply specify none (N/A) if not applicable.*

### CODE REVIEW

File: profile/profiler.go [hunk 1]
```diff
@@ -298,23 +298,24 @@ func (p *Profiler) parseMetric(lines []pmuLine) map[string]float64 {
...
+	return true
 }
 
 func isPerfInstalled() bool {
-	cmd := exec.Command("perf", "--version")
-	b, err := cmd.Output()
+	_, err := exec.LookPath("perf")
 	if err != nil {
-		log.Error(err)
+		log.Printf("perf is not installed or not found in PATH: %v", err)
```
Comment (leokondrashov) [inline profile/profiler.go:315]: Use `Error` for that

File: .github/workflows/unit_tests.yml [hunk 2]
```diff
@@ -22,7 +22,7 @@ env:
 jobs:
   unit-test:
     name: Unit test
-    runs-on: ubuntu-20.04
+    runs-on: [self-hosted, Linux, X64, utest]
```
Comment (leokondrashov) [inline .github/workflows/unit_tests.yml:25]: Do we need to move all of the unit tests on self-hosted?

**Review by leokondrashov — CHANGES_REQUESTED** [review]

File: profile/profiler.go (hunk 1 above)
Comment (lrq619) [reply profile/profiler.go:315]: OK, will change to `Errorf`

File: .github/workflows/unit_tests.yml (hunk 2 above)
Comment (lrq619) [reply .github/workflows/unit_tests.yml:25]: I think so, all jobs under `unit-test` include the step `Install PMU tools`

Comment (leokondrashov) [reply .github/workflows/unit_tests.yml:25]: Yeah, but only the profile unit test actually needs it. You've tried that in separate PR. I'm thinking about offloading to self-hosted only the parts that cannot be ran on github runners. Just like firecracker unit tests in the same file.

Comment (lrq619) [reply .github/workflows/unit_tests.yml:25]: Okay, no problem. I will change the yaml structure

**Review by leokondrashov — APPROVED** [review]
lgtm

**Pull request merged**